plugins/scaffolder-backend/api-report.md

```diff
@@ -333,6 +333,8 @@ export const createPublishGitlabMergeRequestAction: (options: {
   branchName: string;
   targetPath: string;
   token?: string | undefined;
+  removeSourceBranch?: boolean | undefined;
+  gitlabAction?: 'update' | 'create' | 'delete' | undefined;
```

Comment:
What does this do exactly? Is it switch for how all files should be applied in the PR? Wondering why this is needed, does the GitLab API request fail if you include a file that already exists without explicitly setting `'update'`? I'm particularly wondering what `'delete'` looks like from the template creators point of view?

Reply:
Hi @Rugvip Yes this is correct. The GitLab API fails if you try and edit an existing file as it says it already exists due to the action being hardcoded to create [here](https://github.com/backstage/backstage/blob/master/plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/gitlabMergeRequest.ts#L148). I wanted the 3 use cases: 
1. Onboard new user (new files so use 'create' gitlab action which works in the current functionality)
2. Amend Current User Details ('update' existing files which fails currently) 
3. Off-board user ('delete' existing file which fails currently). 
This new option will allow template creators to select which gitlab action they want to use for their templates and the GitLab MR will act accordingly. 
I thought I'd combine the two together as this will help us massively.